tools/run-dev

```diff
@@ -306,6 +307,27 @@ def print_listeners() -> None:
     print()
 
 
+def https_log_filter(record: logging.LogRecord) -> bool:
+    # aiohttp emits an exception with a traceback when receiving an
+    # https request (https://github.com/aio-libs/aiohttp/issues/8065).
+    # Abreviate it to a one-line message.
+    if (
+        record.exc_info is not None
+        and isinstance(error := record.exc_info[1], BadStatusLine)
+        and error.message.startswith(
+            (
+                "Invalid method encountered:\n\n  b'\\x16",
+                'Invalid method encountered:\n\n  b"\\x16',
+            )
+        )
+    ):
+        record.msg = "Rejected https request (this development server only supports http)"
+        record.exc_info = None
```

Comment:
We could suppress the message entirely by returning `False`, but I figure it’s useful to provide some feedback if the user accidentally used https or their browser upgraded the request automatically.

Reply:
Yeah that seems reasonable.

Reply:
We could have the parameters used when running via puppeteer include an option to fully suppress; that might help avoid confusion for folks running those tests?